clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookDownloadService.ts

```diff
@@ -12,552 +12,120 @@
-    errors: Set<string>,
-    startTime: number,
-  ) {
-    {
-      await this.handleQueue(element, errors, progress);
-
-      if (this.queue.length === 0 && errors.size > 0) {
-        errors.clear();
-      }
-
-      if (this.queue.length === 0) {
-        TelemetryService.registerEvent(
-          "Download copybooks from MF",
-          ["copybook", "COBOL", experimentTag],
-          "total time to search copybooks on MF",
-          new Map().set(
-            "time elapsed",
-            TelemetryService.calculateTimeElapsed(startTime, Date.now()),
-          ) as unknown as TelemetryEventMeasurements,
-        );
-      }
-    }
-  }
-
-  private async run(
-    element: CopybookProfile,
-    errors: Set<string>,
-    startTime: number,
-  ) {
     await vscode.window.withProgress(
       {
         location: vscode.ProgressLocation.Notification,
         title: "Fetching copybooks",
       },
       async (
         progress: vscode.Progress<{ message?: string; increment?: number }>,
-      ) => this.process(progress, element, errors, startTime),
+      ) => {
+        await this.processCopybookDownload(
+          progress,
+          documentUri,
+          copybookNames,
+        );
+      },
     );
-    return startTime;
   }
 
-  private async handleQueue(
-    element: CopybookProfile,
-    errors: Set<string>,
+  private async processCopybookDownload(
     progress: vscode.Progress<{ message?: string; increment?: number }>,
-  ) {
-    const toDownload: CopybookProfile[] = [element];
-    while (this.queue.length > 0) {
-      const downloadItem = await this.queue.pop();
-      if (downloadItem) {
-        toDownload.push(downloadItem);
-      }
-    }
-    toDownload.map((cp) => cp.getCopybook()).forEach((cb) => errors.add(cb));
-    CopybookDownloadService.totalDownload = toDownload.length;
-    CopybookDownloadService.completedDownload = 0;
-    try {
-      await this.handleCopybooks(toDownload, errors, progress, false);
+    documentUri: string,
+    copybookNames: CopybookName[],
+  ): Promise<void> {
+    const totalCopybooksToDownload = copybookNames.length;
+    let downloadedCopybooks = 0;
+
+    await Promise.all(
+      copybookNames.map(async (copybookName) => {
+        const isDownloadSuccess = await this.downloadResolver!.downloadCopybook(
+          copybookName,
+          documentUri,
+          undefined,
+        );
 
-      const toDownloadUSS = toDownload
-        .filter((cp) => errors.has(cp.getCopybook()))
-        .map((cp) => cp);
-      const quiteModeOffCopybooks = toDownloadUSS
-        .filter((cp) => !cp.quiet)
-        .map((cp) => cp.getCopybook());
-      errors.forEach((ele) => {
-        if (!quiteModeOffCopybooks.includes(ele)) {
-          errors.delete(ele);
+        if (isDownloadSuccess) {
+          downloadedCopybooks++;
+          this.updateDownloadProgress(
+            progress,
+            totalCopybooksToDownload,
+            downloadedCopybooks,
```

Comment:
should the progress bar be updated regardless of the success of the operation?

Reply:
Ahhaa, yes. Will update and rename downloadedCopybooks variable

Reply:
updated